### PR TITLE
feat: add filtering capabilities to search

### PR DESCRIPTION
**Pull request by nunogois**
## About the changes
https://trello.com/c/Z8Gl0Q9u

### Important files
 - `useSearch.ts`
 - `FeatureToggleListTable.tsx`

### CODE REVIEW

File: src/hooks/useSearch.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    fieldValue.toLowerCase().includes(value.toLowerCase());
+
+export const getSearchText = (searchValue: string) =>
+    searchValue
+        .split(' ')
+        .filter(fragment => !isValidFilter(fragment, '\\w+'))
+        .join(' ');
+
+const isValidFilter = (input: string, match: string) =>
+    new RegExp(`${match}:\\w+`).test(input);
```
[the remark this answers is not in the document]
Comment (nunogois) [reply src/hooks/useSearch.ts:79]: @sonatype-lift ignore

File: src/component/common/Table/TableSearch/TableSearchField/TableSearchFieldSuggestions/TableSearchFieldSuggestions.tsx [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+import {
+    getColumnValues,
+    getFilterableColumns,
+    getFilterValues,
+    getSearchTextGenerator,
+    IGetSearchContextOutput,
+} from 'hooks/useSearch';
+import { useMemo, VFC } from 'react';
+
+const randomIndex = (arr: any[]) => Math.floor(Math.random() * arr.length);
```
[the remark this answers is not in the document]
Comment (nunogois) [reply src/component/common/Table/TableSearch/TableSearchField/TableSearchFieldSuggestions/TableSearchFieldSuggestions.tsx:13]: @sonatype-lift ignore

File: src/component/common/Table/TableSearch/TableSearchField/TableSearchFieldSuggestions/TableSearchFieldSuggestions.tsx [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+const StyledPaper = styled(Paper)(({ theme }) => ({
```
Comment (FredrikOseberg) [inline src/component/common/Table/TableSearch/TableSearchField/TableSearchFieldSuggestions/TableSearchFieldSuggestions.tsx:15]: I think according to our discussion earlier these should be at the bottom of the file? Or did I misunderstand?

File: src/component/common/Table/TableSearch/TableSearchField/TableSearchFieldSuggestions/TableSearchFieldSuggestions.tsx [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+                <StyledFilterList />
+                <Box>
+                    <ConditionallyRender
+                        condition={Boolean(searchContext.searchValue)}
+                        show={
+                            <>
+                                <StyledHeader>Searching for:</StyledHeader>
+                                <ConditionallyRender
+                                    condition={Boolean(searchText)}
+                                    show={
```
Comment (FredrikOseberg) [inline src/component/common/Table/TableSearch/TableSearchField/TableSearchFieldSuggestions/TableSearchFieldSuggestions.tsx:124]: This render function is getting quite large. I think we can simplify this one by breaking some of it into subcomponents.

**Review by FredrikOseberg — CHANGES_REQUESTED** [review]
Awesome! Some minor details left, and we'll need @NicolaeUnleash input on some of the UX details here, but I think this can be merged soon.

File: src/component/common/Table/TableSearch/TableSearchField/TableSearchFieldSuggestions/TableSearchFieldSuggestions.tsx (hunk 3 above)
Comment (Tymek) [reply src/component/common/Table/TableSearch/TableSearchField/TableSearchFieldSuggestions/TableSearchFieldSuggestions.tsx:15]: I thinks so. I have a preference for putting it on top, to make sure main exported thing is always last. If the file is getting too long, I would put styled in separate one separate file.

File: src/component/common/Table/TableSearch/TableSearchField/TableSearchFieldSuggestions/TableSearchFieldSuggestions.tsx (hunk 4 above)
Comment (nunogois) [reply src/component/common/Table/TableSearch/TableSearchField/TableSearchFieldSuggestions/TableSearchFieldSuggestions.tsx]: Addressed in: https://github.com/Unleash/unleash-frontend/pull/1052/commits/3cc1535b3bbf991375c890a4397553481f1072c6

**Review by FredrikOseberg — APPROVED** [review]
LGTM

**Pull request merged**